perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -1744,15 +1760,34 @@ public Member expireMember(PerunSession sess, Member member) throws WrongReferen
 				List<Vo> parentVos = getPerunBl().getVosManagerBl().getParentVos(sess, member.getVoId());
 				for (Vo vo : parentVos) {
 					try {
-						Member newMember = perunBl.getMembersManagerBl().createMember(sess, vo, user);
-						perunBl.getMembersManagerBl().validateMember(sess, newMember);
-						addMemberToParentVosGroups(sess, member);
-						Attribute attribute = perunBl.getAttributesManagerBl().getAttribute(sess, newMember, A_MEMBER_DEF_MEMBER_ORGANIZATIONS);
-						attribute.setValue(new ArrayList<String>());
-						perunBl.getAttributesManagerBl().setAttribute(sess, member, attribute);
-					} catch (ExtendMembershipException  | WrongReferenceAttributeValueException | AttributeNotExistsException | WrongAttributeAssignmentException e) {
+						Vo memberVo = perunBl.getVosManagerBl().getVoById(sess, member.getVoId());
+						try {
+							Member newMember = perunBl.getMembersManagerBl().createMember(sess, vo, user);
+							perunBl.getMembersManagerBl().validateMember(sess, newMember);
+							addMemberToParentVosGroups(sess, member);
+
+							// set memberOrganizations attribute to empty list
+							Attribute memberOrganizations = perunBl.getAttributesManagerBl().getAttribute(sess, newMember, A_MEMBER_DEF_MEMBER_ORGANIZATIONS);
+							memberOrganizations.setValue(new ArrayList<String>());
+							perunBl.getAttributesManagerBl().setAttribute(sess, newMember, memberOrganizations);
+
+							// update memberOrganizationsHistory attribute to contain VO
+							Attribute memberOrganizationsHistory = perunBl.getAttributesManagerBl().getAttribute(sess, newMember, A_MEMBER_DEF_MEMBER_ORGANIZATIONS_HISTORY);
+							memberOrganizationsHistory.setValue(new ArrayList<>(List.of(memberVo.getShortName())));
+							perunBl.getAttributesManagerBl().setAttribute(sess, newMember, memberOrganizationsHistory);
+						} catch (AlreadyMemberException e) {
+							// update memberOrganizationsHistory attribute to contain VO
+							Attribute memberOrganizationsHistory = perunBl.getAttributesManagerBl().getAttribute(sess, member, A_MEMBER_DEF_MEMBER_ORGANIZATIONS_HISTORY);
```

Comment:
One last thing - we don't want to set this attribute for `member` but for his equivalent in parentVo. There is a method `getMemberByUserId(sess, vo, member.getUserId())` which should fetch such member.